core/commands/storage.go

```diff
@@ -146,18 +155,11 @@ var storageUploadInitCmd = &cmds.Command{
 		clientConn, err := ledger.LedgerConnection()
 		defer ledger.CloseConnection(clientConn)
```

Comment:
Connection opening/closing should be part of `getChannelInfo`, we don't want to hold long connections/resources until all the p2p calls complete (could take a long time)

Reply:
my original idea is leaving ledger client outside. Since whenever you want to get information in channel you have to establish connection and have a ledger client first, if each process needs updated info from ledger and that will be open/close, reopen/reclose, again and again, will it be time cost there? But I agree it is less secure if we open the channel for a long time.

Reply:
Makes sense if we don't have p2pcall's in between client connections... for this flow it's a long connection